File: src/main/java/com/tms/csp/parse/VarTokenizerStd.java
package com.tms.csp.parse;

import com.tms.csp.parse.std.VarTokenStd;

public class VarTokenizerStd extends VarTokenizer {

    public boolean isValidVarFirstChar(char c) {
        return Character.isLetterOrDigit(c);
    }

    public boolean isValidVarChar(char c) {
        return Character.isLetterOrDigit(c) || c == UNDERSCORE || c == '|' || c == '.' || c == ':';
    }


    @Override
    public Token matches(Stream stream) {
        if (!matchesInternal(stream)) return null;
        StringBuilder sb = new StringBuilder();

        while (keepReading(stream)) {
            sb.append(stream.take());
        }
        return new VarTokenStd(sb.toString());
    }


    public boolean matchesInternal(Stream stream) {
        char c = stream.peek();
        boolean b = isValidVarFirstChar(c);
        if (!b) {
            System.err.println("badFirstChar: " + c + " at cursor[" + stream.getCursor() +"] formula string: " + stream.getInput() );
        }
        return b;
    }

    public boolean keepReading(Stream stream) {
        if (stream.isEol()) {
            return false;
        }
        char c = stream.peek();

        boolean validVarChar = isValidVarChar(c);

        if (!validVarChar) {
//            System.err.println("invalidVarChar[" + c + "]");
        }

        return validVarChar;
    }


    public boolean isValidVarFirstChar(Stream stream) {
        return isValidVarFirstChar(stream.peek());
    }

}

File: src/main/java/com/smartsoft/csp/parse/VarTokenizerStd.kt
package com.smartsoft.csp.parse

class VarTokenizerStd : VarTokenizer() {

    fun isValidVarFirstChar(c: Char): Boolean {
        return Character.isLetterOrDigit(c)
    }

    fun isValidVarChar(c: Char): Boolean {
        return Character.isLetterOrDigit(c) || c == UNDERSCORE || c == '|' || c == '.' || c == ':'
    }


    override fun matches(stream: Stream): Token? {
        if (!matchesInternal(stream)) return null
        val sb = StringBuilder()

        while (keepReading(stream)) {
            sb.append(stream.take())
        }
        return VarTokenStd(sb.toString())
    }


    fun matchesInternal(stream: Stream): Boolean {
        val c = stream.peek()
        val b = isValidVarFirstChar(c)
        if (!b) {
            System.err.println("badFirstChar: " + c + " at cursor[" + stream.getCursor() + "] formula string: " + stream.input)
        }
        return b
    }

    fun keepReading(stream: Stream): Boolean {
        if (stream.isEol) {
            return false
        }
        val c = stream.peek()

        val validVarChar = isValidVarChar(c)

        if (!validVarChar) {
            //            System.err.println("invalidVarChar[" + c + "]");
        }

        return validVarChar
    }


    fun isValidVarFirstChar(stream: Stream): Boolean {
        return isValidVarFirstChar(stream.peek())
    }

}
